File: sdk/examples/typescript/remote-connect.ts
import { RoderAgent } from "../../typescript/src/index.js";

if (process.env.RODER_SDK_LIVE !== "1") {
  console.log("Set RODER_SDK_LIVE=1, RODER_REMOTE_URL, and RODER_REMOTE_TOKEN to run this example.");
  process.exit(0);
}

const url = process.env.RODER_REMOTE_URL;
const token = process.env.RODER_REMOTE_TOKEN;
if (!url || !token) {
  throw new Error("RODER_REMOTE_URL and RODER_REMOTE_TOKEN are required");
}

const agent = await RoderAgent.create({ remote: { url, token } });
console.log(await agent.listProviders());
await agent.close();

File: sdk/examples/python/remote_connect.py
from __future__ import annotations

import os

import anyio

from roder_sdk import RoderAgent


async def main() -> None:
    if os.environ.get("RODER_SDK_LIVE") != "1":
        print("Set RODER_SDK_LIVE=1, RODER_REMOTE_URL, and RODER_REMOTE_TOKEN to run this example.")
        return
    url = os.environ["RODER_REMOTE_URL"]
    token = os.environ["RODER_REMOTE_TOKEN"]
    agent = await RoderAgent.create(remote={"url": url, "token": token})
    print(await agent.list_providers())
    await agent.close()


anyio.run(main)
